coding-guides/using-audit-logs.md

```diff
@@ -0,0 +1,142 @@
+let it be viewing them for certain actions to adding things into the audit logs. 
+This guide will explain everything about audit logs and how to use them.
+
+The first thing that you will need is a working Discord Bot. If you do not have one, please visit [Your First Bot](getting-started/your-basic-bot.md) to get started.
+
+Now, some things to take note of. This guide is using discord.js@11.3.2 The bot will need some permissions. The main permission the bot will need is `'VIEW_AUDIT_LOGS'`. This permission allows the bot to view the audit logs and do whatever you want to, obeying the Discord ToS that is. 
+
+Now that the permission has been established. Lets get started!
+
+Firstly, we need to know what we are doing with the audit logs. 
+Lets log who deleted a message using the messageDeleted event. This event will fire whenever a cached message is deleted in a server.
+
+```js
+client.on('messageDeleted', async (message) => {
+  // Firstly, we need a logs channel. 
+  const logs = message.guild.channels.find('name', 'logs');
+  
+  // If there is no logs channel, we can create it if we have the 'MANAGE_CHANNELS' permission
+  // Remember, this is completely options. Use to your best judgement.
+  if (message.guild.me.hasPermission('MANAGE_CHANNELS') && !logs) {
+    message.guild.createChannel('logs', 'text');
+  }
+  
+  // If we do not have permissions, console log both errors
+  if (!message.guild.me.hasPermission('MANAGE_CHANNELS') && !logs) { 
+    console.log('The logs channel does not exist and tried to create the channel but I am lacking permissions')
+  }
+  
+  // Now that we have the logs channel created...hopefully...we can send messages to it
+  // before we do that, lets establish who deleted the message
+  // The "type" is how you will be searching through the audit logs. Like roles updated or members banned.
+  const executor = await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(audit => audit.entries.first())
+  
+  
+})
+```
+Lets take a break to explain exactly whats going on. The executor will be retrieving the very latest audit log entry and all of its information that goes with it. What does this information contain? Everything we need for logging. 
+
+```js
+GuildAuditLogsEntry {
+  targetType: 'MESSAGE',
+  actionType: 'DELETE',
+  action: 'MESSAGE_DELETE',
+  reason: null,
+  executor: [Object],
+  changes: null,
+  id: '434116792567201792',
+  extra: [Object],
+  target: [Object] }
+```
+Notice the `reason` field. Some audit logs, like kicking and banning, can provide a reason. You can probably make logs of when a user is banned and for whatever reason.
+What we want is the executor of the action. We do that by going to the `executor` target as that is where the user is stored.
+So, our `executor` should be look similar to this:
+
+```js
+  const executor = await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(audit => audit.entries.first().executor.username)
```

Comment:
Generally speaking, people will actually want to use more than just the username of an audit log. It's best practice to actually return the whole entry and then take a property *from* it. that would make more sense.